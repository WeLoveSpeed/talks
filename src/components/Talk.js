import React, { Component } from 'react';
import moment from 'moment';
import md5 from 'js-md5';
import { isPK } from '../services/FormatService';
import { List, ListItem } from 'material-ui/List';
import Avatar from 'material-ui/Avatar';
import CommunicationEmail from 'material-ui/svg-icons/communication/email';
import {red500, lightBlack, orange500, teal500} from 'material-ui/styles/colors';
import Chip from 'material-ui/Chip';

class Talk extends Component {
  getDate(date) {
    return moment(date, 'DD/MM/YYYY').format('DD/MM/YYYY');
  }

  getText(text) {
    if (!text) {
      return false;
    }

    const paragraphs = text.match(/[^\r\n]+/g);
    if (paragraphs === null) {
      return false;
    }
    return paragraphs.map((p, key) => <p key={key}>{p}</p>)
  }

  getFormat(format) {
    if (format === 'Pecha Kucha : 20 images x 20 secondes') {
      return <Avatar backgroundColor={red500}>PK</Avatar>
    }
    if (format === 'Lightning Talk : 5 minutes') {
      return <Avatar backgroundColor={orange500}>LT</Avatar>
    }

    return null;
  }

  getAvatar(email) {
    const hash = md5(email.toLowerCase());
    if (email) {
      return <Avatar src={`https://www.gravatar.com/avatar/${hash}`} />
    }

    return null;
  }

  sendMail(email) {
    window.location = `mailto:${email}`;
  }

  render() {
    const {talk} = this.props;

    if (talk === null) {
      return null;
    }

    const formatStyle = {
      backgroundColor: isPK(talk.formats) ? teal500 : orange500,
      marginTop: '.85em'
    }

    return (
      <section className="Talk">
        <h2>{talk.titre_de_ta_presentation}</h2>
        <Chip labelColor='white' style={formatStyle}>{talk.formats}</Chip>
        <List>
          <ListItem
            onTouchTap={() => this.sendMail(talk.email)}
            primaryText={
              <span>{talk.prenom_et_nom}<span style={{color: lightBlack}}>, le {this.getDate(talk.timestamp)}</span></span>
            }
            secondaryText={talk.email}
            leftAvatar={this.getAvatar(talk.email)}
            rightIcon={<CommunicationEmail color={red500} />}
            />
        </List>
        <h3>Description</h3>
        {this.getText(talk.description_de_ta_presentation)}


        <h3>Si le public ne devait retenir qu'une chose, ce serait...</h3>
        {this.getText(talk.si_le_public_ne_devait_retenir_quune_chose_ce_serait)}

        <h3>Tu veux ajouter quelque chose ?</h3>
        {this.getText(talk.tu_veux_ajouter_quelque_chose)}
      </section>
    );
  }
}

export default Talk;
