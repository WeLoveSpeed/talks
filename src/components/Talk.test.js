import React, { Component } from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Talk from "./Talk";

const mockTalk = {
  prenom_first_name: "Boris",
  nom_last_name: "Schapira",
  courriel_e_mail: "test@test.com",
  titre_title: "Wesh, bien ou bien ?",
  description: "Ceci est la description.",
  url_de_profil_principal_main_profile_url: "https://twitter.com/_welovespeed",
  url_dune_photo_de_vous_profile_picture_url: "https://pbs.twimg.com/profile_images/1049198944826417152/w4Vfcyja_400x400.jpg",
  horodateur: "12/03/2019 16:04:23",
  duree_duration: "45 minutes",
  type_dintervention_talk_level: "Approfondissement / Deep Dive",
  langue_language: "FranÃ§ais",
  un_message_a_laisser_a_lequipe_organisatrice_do_you_have_a_message_for_the_organizing_team: "Ceci est le message.",
  where_will_you_come_from: "Outta here."
};

describe("Talk component", () => {
  it("should renders correctly", () => {
    const component = shallow(<Talk talk={mockTalk} />);
    expect(component).toMatchSnapshot();
  });
});
