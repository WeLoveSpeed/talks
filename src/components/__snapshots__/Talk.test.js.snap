exports[`Talk component should renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="Wrapper">
    <div
        className="Talk">
        <h2>
            Wesh, bien ou bien ?
        </h2>
        <Chip
            labelColor="white"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyboardFocus={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
                Object {
                    "backgroundColor": "#ff9800",
                    "marginTop": ".85em",
                  }
            } />
        <Connect(NotesView) />
        <Divider
            inset={false} />
        <List>
            <ListItem
                autoGenerateNestedIndicator={true}
                disableKeyboardFocus={false}
                disabled={true}
                initiallyOpen={false}
                insetChildren={false}
                leftAvatar={
                    <Avatar
                        size={40}
                        src="https://pbs.twimg.com/profile_images/1049198944826417152/w4Vfcyja_400x400.jpg" />
                }
                nestedItems={Array []}
                nestedLevel={0}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onNestedListToggle={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                open={null}
                primaryText={
                    <span>
                        Boris Schapira
                        <span
                            style={
                                Object {
                                    "color": "rgba(0, 0, 0, 0.54)",
                                  }
                            }>
                            , le 
                            12/03/2019
                        </span>
                    </span>
                }
                primaryTogglesNestedList={false}
                rightIcon={
                    <IconButton
                        disableTouchRipple={false}
                        disabled={false}
                        iconStyle={Object {}}
                        onClick={[Function]}
                        style={
                            Object {
                                "padding": 0,
                                "width": 32,
                              }
                        }
                        tooltipPosition="top-right"
                        touch={true}>
                        <CommunicationEmail
                            color="#f44336" />
                    </IconButton>
                }
                secondaryText="test@test.com"
                secondaryTextLines={1} />
        </List>
        <Divider
            inset={false} />
        <h3>
            Approfondissement / Deep Dive
             (
            45 minutes
            )
        </h3>
        <p>
            Ceci est la description.
        </p>
        <h3>
            Message pour l'équipe
        </h3>
        <p>
            Ceci est le message.
        </p>
    </div>
</section>,
  "nodes": Array [
    <section
      className="Wrapper">
      <div
            className="Talk">
            <h2>
                  Wesh, bien ou bien ?
            </h2>
            <Chip
                  labelColor="white"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyboardFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={
                        Object {
                              "backgroundColor": "#ff9800",
                              "marginTop": ".85em",
                            }
                  } />
            <Connect(NotesView) />
            <Divider
                  inset={false} />
            <List>
                  <ListItem
                        autoGenerateNestedIndicator={true}
                        disableKeyboardFocus={false}
                        disabled={true}
                        initiallyOpen={false}
                        insetChildren={false}
                        leftAvatar={
                              <Avatar
                                    size={40}
                                    src="https://pbs.twimg.com/profile_images/1049198944826417152/w4Vfcyja_400x400.jpg" />
                        }
                        nestedItems={Array []}
                        nestedLevel={0}
                        onKeyboardFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onNestedListToggle={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        open={null}
                        primaryText={
                              <span>
                                    Boris Schapira
                                    <span
                                          style={
                                                Object {
                                                      "color": "rgba(0, 0, 0, 0.54)",
                                                    }
                                          }>
                                          , le 
                                          12/03/2019
                                    </span>
                              </span>
                        }
                        primaryTogglesNestedList={false}
                        rightIcon={
                              <IconButton
                                    disableTouchRipple={false}
                                    disabled={false}
                                    iconStyle={Object {}}
                                    onClick={[Function]}
                                    style={
                                          Object {
                                                "padding": 0,
                                                "width": 32,
                                              }
                                    }
                                    tooltipPosition="top-right"
                                    touch={true}>
                                    <CommunicationEmail
                                          color="#f44336" />
                              </IconButton>
                        }
                        secondaryText="test@test.com"
                        secondaryTextLines={1} />
            </List>
            <Divider
                  inset={false} />
            <h3>
                  Approfondissement / Deep Dive
                   (
                  45 minutes
                  )
            </h3>
            <p>
                  Ceci est la description.
            </p>
            <h3>
                  Message pour l'équipe
            </h3>
            <p>
                  Ceci est le message.
            </p>
      </div>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Talk
        talk={
                Object {
                        "courriel_e_mail": "test@test.com",
                        "description": "Ceci est la description.",
                        "duree_duration": "45 minutes",
                        "horodateur": "12/03/2019 16:04:23",
                        "langue_language": "Français",
                        "nom_last_name": "Schapira",
                        "prenom_first_name": "Boris",
                        "titre_title": "Wesh, bien ou bien ?",
                        "type_dintervention_talk_level": "Approfondissement / Deep Dive",
                        "un_message_a_laisser_a_lequipe_organisatrice_do_you_have_a_message_for_the_organizing_team": "Ceci est le message.",
                        "url_de_profil_principal_main_profile_url": "https://twitter.com/_welovespeed",
                        "url_dune_photo_de_vous_profile_picture_url": "https://pbs.twimg.com/profile_images/1049198944826417152/w4Vfcyja_400x400.jpg",
                        "where_will_you_come_from": "Outta here.",
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Talk {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "talk": Object {
            "courriel_e_mail": "test@test.com",
            "description": "Ceci est la description.",
            "duree_duration": "45 minutes",
            "horodateur": "12/03/2019 16:04:23",
            "langue_language": "Français",
            "nom_last_name": "Schapira",
            "prenom_first_name": "Boris",
            "titre_title": "Wesh, bien ou bien ?",
            "type_dintervention_talk_level": "Approfondissement / Deep Dive",
            "un_message_a_laisser_a_lequipe_organisatrice_do_you_have_a_message_for_the_organizing_team": "Ceci est le message.",
            "url_de_profil_principal_main_profile_url": "https://twitter.com/_welovespeed",
            "url_dune_photo_de_vous_profile_picture_url": "https://pbs.twimg.com/profile_images/1049198944826417152/w4Vfcyja_400x400.jpg",
            "where_will_you_come_from": "Outta here.",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="Wrapper">
          <div
                    className="Talk">
                    <h2>
                              Wesh, bien ou bien ?
                    </h2>
                    <Chip
                              labelColor="white"
                              onBlur={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyboardFocus={[Function]}
                              onMouseDown={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              style={
                                        Object {
                                                  "backgroundColor": "#ff9800",
                                                  "marginTop": ".85em",
                                                }
                              } />
                    <Connect(NotesView) />
                    <Divider
                              inset={false} />
                    <List>
                              <ListItem
                                        autoGenerateNestedIndicator={true}
                                        disableKeyboardFocus={false}
                                        disabled={true}
                                        initiallyOpen={false}
                                        insetChildren={false}
                                        leftAvatar={
                                                  <Avatar
                                                            size={40}
                                                            src="https://pbs.twimg.com/profile_images/1049198944826417152/w4Vfcyja_400x400.jpg" />
                                        }
                                        nestedItems={Array []}
                                        nestedLevel={0}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onNestedListToggle={[Function]}
                                        onTouchEnd={[Function]}
                                        onTouchStart={[Function]}
                                        open={null}
                                        primaryText={
                                                  <span>
                                                            Boris Schapira
                                                            <span
                                                                      style={
                                                                                Object {
                                                                                          "color": "rgba(0, 0, 0, 0.54)",
                                                                                        }
                                                                      }>
                                                                      , le 
                                                                      12/03/2019
                                                            </span>
                                                  </span>
                                        }
                                        primaryTogglesNestedList={false}
                                        rightIcon={
                                                  <IconButton
                                                            disableTouchRipple={false}
                                                            disabled={false}
                                                            iconStyle={Object {}}
                                                            onClick={[Function]}
                                                            style={
                                                                      Object {
                                                                                "padding": 0,
                                                                                "width": 32,
                                                                              }
                                                            }
                                                            tooltipPosition="top-right"
                                                            touch={true}>
                                                            <CommunicationEmail
                                                                      color="#f44336" />
                                                  </IconButton>
                                        }
                                        secondaryText="test@test.com"
                                        secondaryTextLines={1} />
                    </List>
                    <Divider
                              inset={false} />
                    <h3>
                              Approfondissement / Deep Dive
                               (
                              45 minutes
                              )
                    </h3>
                    <p>
                              Ceci est la description.
                    </p>
                    <h3>
                              Message pour l'équipe
                    </h3>
                    <p>
                              Ceci est le message.
                    </p>
          </div>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="Wrapper">
          <div
                    className="Talk">
                    <h2>
                              Wesh, bien ou bien ?
                    </h2>
                    <Chip
                              labelColor="white"
                              onBlur={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyboardFocus={[Function]}
                              onMouseDown={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              style={
                                        Object {
                                                  "backgroundColor": "#ff9800",
                                                  "marginTop": ".85em",
                                                }
                              } />
                    <Connect(NotesView) />
                    <Divider
                              inset={false} />
                    <List>
                              <ListItem
                                        autoGenerateNestedIndicator={true}
                                        disableKeyboardFocus={false}
                                        disabled={true}
                                        initiallyOpen={false}
                                        insetChildren={false}
                                        leftAvatar={
                                                  <Avatar
                                                            size={40}
                                                            src="https://pbs.twimg.com/profile_images/1049198944826417152/w4Vfcyja_400x400.jpg" />
                                        }
                                        nestedItems={Array []}
                                        nestedLevel={0}
                                        onKeyboardFocus={[Function]}
                                        onMouseEnter={[Function]}
                                        onMouseLeave={[Function]}
                                        onNestedListToggle={[Function]}
                                        onTouchEnd={[Function]}
                                        onTouchStart={[Function]}
                                        open={null}
                                        primaryText={
                                                  <span>
                                                            Boris Schapira
                                                            <span
                                                                      style={
                                                                                Object {
                                                                                          "color": "rgba(0, 0, 0, 0.54)",
                                                                                        }
                                                                      }>
                                                                      , le 
                                                                      12/03/2019
                                                            </span>
                                                  </span>
                                        }
                                        primaryTogglesNestedList={false}
                                        rightIcon={
                                                  <IconButton
                                                            disableTouchRipple={false}
                                                            disabled={false}
                                                            iconStyle={Object {}}
                                                            onClick={[Function]}
                                                            style={
                                                                      Object {
                                                                                "padding": 0,
                                                                                "width": 32,
                                                                              }
                                                            }
                                                            tooltipPosition="top-right"
                                                            touch={true}>
                                                            <CommunicationEmail
                                                                      color="#f44336" />
                                                  </IconButton>
                                        }
                                        secondaryText="test@test.com"
                                        secondaryTextLines={1} />
                    </List>
                    <Divider
                              inset={false} />
                    <h3>
                              Approfondissement / Deep Dive
                               (
                              45 minutes
                              )
                    </h3>
                    <p>
                              Ceci est la description.
                    </p>
                    <h3>
                              Message pour l'équipe
                    </h3>
                    <p>
                              Ceci est le message.
                    </p>
          </div>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Talk
    talk={
        Object {
            "courriel_e_mail": "test@test.com",
            "description": "Ceci est la description.",
            "duree_duration": "45 minutes",
            "horodateur": "12/03/2019 16:04:23",
            "langue_language": "Français",
            "nom_last_name": "Schapira",
            "prenom_first_name": "Boris",
            "titre_title": "Wesh, bien ou bien ?",
            "type_dintervention_talk_level": "Approfondissement / Deep Dive",
            "un_message_a_laisser_a_lequipe_organisatrice_do_you_have_a_message_for_the_organizing_team": "Ceci est le message.",
            "url_de_profil_principal_main_profile_url": "https://twitter.com/_welovespeed",
            "url_dune_photo_de_vous_profile_picture_url": "https://pbs.twimg.com/profile_images/1049198944826417152/w4Vfcyja_400x400.jpg",
            "where_will_you_come_from": "Outta here.",
          }
    } />,
}
`;
